File: 107-binary-tree-level-order-traversal-ii/107-binary-tree-level-order-traversal-ii.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
		Queue<TreeNode> q = new LinkedList<>();

		List<List<Integer>> return_list = new LinkedList<List<Integer>>();

		if (root == null) {
			return return_list;
		}

		q.add(root);
		
		while(!q.isEmpty()) {
			int size = q.size();
			List<Integer> temp_list = new LinkedList<Integer>();
			
			for(int i = 0; i < size; i++) {
				TreeNode currentNode = q.poll();
				temp_list.add(currentNode.val);
				
				if(currentNode.left != null)
					q.offer(currentNode.left);
				if(currentNode.right != null)
					q.offer(currentNode.right);
			}
			return_list.add(0, temp_list);
		}
		return return_list;	
	}
}

File: 107-binary-tree-level-order-traversal-ii/107-binary-tree-level-order-traversal-ii.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrderBottom(self, root: Optional[TreeNode]) -> List[List[int]]:
            if(root == None):
                return []
            
            queue = []
            queue.append(root)

            list = []

            while(len(queue) != 0):
                size = len(queue)

                temp = []
                for i in range(size):
                    current = queue.pop(0)
                    temp.append(current.val)

                    # print(current.val, end = ' ')

                    if(current.left != None):
                        queue.append(current.left)
                    if(current.right != None):
                        queue.append(current.right)

                list.insert(0, temp)

                print('\n')

            return list